models/repo_collaboration.go

```diff
@@ -186,9 +186,49 @@ func (repo *Repository) DeleteCollaboration(uid int64) (err error) {
 		return err
 	}
 
+	// Unassign a user from any issue (s)he has been assigned to in the
+	// repository
+	if err := repo.removeIssueAssignees(sess, uid); err != nil {
+		return err
+	}
+
 	return sess.Commit()
 }
 
+func (repo *Repository) removeIssueAssignees(e Engine, userID int64) error {
+
+	user, err := GetUserByID(userID)
+	if err != nil {
+		return err
+	}
+
+	perm, err := getUserRepoPermission(e, repo, user)
+	if err != nil {
+		return err
+	}
+
+	if perm.CanWrite(UnitTypeIssues) {
```

Comment:
This check (`if perm.CanWrite(UnitTypeIssues)`) is not right. `CanBeAssigned()` is the only check to do here.